.github/workflows/tests.yml

```diff
@@ -53,16 +53,105 @@ jobs:
         run: cargo test --all-targets
 
+  aws-lc-rs-1804-gcc:
+    if: github.repository_owner == 'aws'
+    name: GCC ${{ matrix.gcc_version }} - Force CMake ${{ matrix.cmake }}
+    runs-on: ubuntu-20.04
+    container:
+      image: ubuntu:18.04
+    env:
+      AWS_LC_SYS_CMAKE_BUILDER: ${{ matrix.cmake }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake: [ '0', '1' ]
+        gcc_version: [ '4.8', '5', '6' ]
+    steps:
+      - run: |
+          apt-get update
+          apt-get install -y build-essential git cmake curl sudo
+          curl -L -O -J https://github.com/PowerShell/PowerShell/releases/download/v7.2.23/powershell_7.2.23-1.deb_amd64.deb
+          dpkg -i powershell_7.2.23-1.deb_amd64.deb
+          apt-get install -f
+          rm powershell_7.2.23-1.deb_amd64.deb
+      - name: Checkout
+        run: |
+          git config --global --add safe.directory '*'
+          git clone --recursive ${{ github.server_url }}/${{ github.repository }}.git .
+          git fetch origin ${{ github.sha }}
+          git checkout -b ci-job ${{ github.sha }}
+      - uses: dtolnay/rust-toolchain@master
+        id: toolchain
+        with:
+          toolchain: stable
+      - name: Set up GCC
+        uses: egor-tensin/setup-gcc@v1.3
+        with:
+          version: ${{ matrix.gcc_version }}
+          platform: x64
+      - name: Run cargo test (debug)
+        run: cargo test -p aws-lc-rs --all-targets --features unstable
+      - name: Run cargo test (release)
+        run: cargo test -p aws-lc-rs --release --all-targets --features unstable
+
+  aws-lc-rs-2004-gcc:
+    if: github.repository_owner == 'aws'
+    name: GCC ${{ matrix.gcc_version }} - Force CMake ${{ matrix.cmake }}
+    runs-on: ubuntu-20.04
+    env:
+      AWS_LC_SYS_CMAKE_BUILDER: ${{ matrix.cmake }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake: [ '0', '1' ]
+        gcc_version: [ '7', '8' ]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+      - uses: dtolnay/rust-toolchain@master
+        id: toolchain
+        with:
+          toolchain: stable
+      - name: Set up GCC
+        uses: egor-tensin/setup-gcc@v1.3
+        with:
+          version: ${{ matrix.gcc_version }}
+          platform: x64
+      - name: Run cargo test (debug)
+        run: cargo test -p aws-lc-rs --all-targets --features unstable
+      - name: Run cargo test (release)
+        run: cargo test -p aws-lc-rs --release --all-targets --features unstable
+      - if: ${{ matrix.gcc_version == '8' }}
+        name: Verify paths found in debug build
+        run: |
+          DEBUG_LIBCRYPTO=$(find ./target/debug -name "libaws_lc_*_crypto.a")
+          if strings ${DEBUG_LIBCRYPTO} | grep runner; then 
+            exit 0; # SUCCESS
+          else
+            exit 1; # FAIL - we expected to find "runner"  (i.e., a path)
+          fi
+      - if: ${{ matrix.gcc_version == '8' }}
```

Comment:
Is there a reason why we check this specific version?

Reply:
The `-ffile-prefix-map` option (which can strip build environment paths from the resulting binary) is only supported on GCC v8 and later. Our use of this flags was causing failures with people building with CMake and an older compiler.

This tests verifies that the path where the build is being performed (which is under the "runner"s home directory) is not found in the release binary.

Reply:
Ah that makes sense, I didn't realize this test dimension was intertwined with the flag. A comment here might help since the knowledge is a bit specific, but not absolutely required.